### PR TITLE
feature: add continuous when keyboard up/down arrow; trigger metaKey/ctrlKey support configuration

### PR DESCRIPTION
**Pull request by afeiship**
## Feature1: continuous
continuous: true
if arrow up：when `idx ==0`, setIndex will to `items.length - 1`
if arrow down: when `idx == items.length -1`, setIndex will to `0`

continuous: false
old behavior

## Feature2: trigger hotkeys support configuration
```ts
  hotkeys?: {
    metaKey?: boolean;
    ctrlKey?: boolean;
  }
```

### CODE REVIEW

File: src/KBarResults.tsx [hunk 1]
```diff
@@ -59,8 +61,9 @@ export const KBarResults: React.FC<KBarResultsProps> = (props) => {
       ) {
         event.preventDefault();
         query.setActiveIndex((index) => {
+          const target = continuous ? 0 : index;
```
Comment (timc1) [inline src/KBarResults.tsx:64]: If the first item is not part of a group; e.g. the theme action is selected, this logic wouldn't work:

Screen Shot 2021-12-25 at 3 43 45 PM
`

File: src/types.ts [hunk 2]
```diff
@@ -29,6 +29,10 @@ export interface KBarOptions {
     enterMs?: number;
     exitMs?: number;
   };
+  hotkeys?: {
+    metaKey?: boolean;
+    ctrlKey?: boolean;
+  };
```
Comment (timc1) [inline src/types.ts:35]: I think `hotkeys` here is too general – are we talking about hotkeys for triggering the command bar, shortcuts, or in this case the up/down navigation controls?

**Review by timc1 — COMMENTED** [review]
Hey @afeiship! Thanks a lot for the contribution – if possible, please do add an issue for discussion prior to adding new feature. For now, I'd like us to keep the current behavior of not wrapping the active index as there are a few additional edge cases that we need to take into consideration.

File: src/KBarResults.tsx (hunk 1 above)
Comment (afeiship) [reply src/KBarResults.tsx:64]: @timc1 thanks, i will add an issue.

File: src/types.ts (hunk 2 above)
Comment (afeiship) [reply src/types.ts:35]: @timc1 this is my case:
https://github.com/timc1/kbar/issues/148

**Pull request closed without merging**